tests/integration/routes/test_confirmation_email.py

```diff
@@ -123,41 +152,77 @@ def test_default_themed_schema_with_confirmation_email_not_set(self):
-        self.assertInBody("A confirmation email has been sent")
+        # Then I get an error on the confirm email page
+        self.assertEqualPageTitle("Error: Confirm your email address - Census 2021")
+        self.assertInBody("There is a problem with this page")
+        self.assertInBody("Select an answer")
 
-    def test_thank_you_page_confirmation_email_white_space(self):
+    def test_confirm_email_no(self):
         # Given I launch and complete the test_confirmation_email questionnaire
         self._launch_and_complete_questionnaire()
 
-        # When I enter a valid email which has leading and trailing whitespace
-        self.post({"email": " email@example.com "})
+        # When I enter a valid email but answer no on the confirm email page
+        self.post({"email": "email@example.com"})
+        self.post({"confirm-email": "No, I need to change it"})
 
-        # Then I get confirmation that the email has been sent
-        self.assertInUrl("confirmation-email/sent")
-        self.assertInBody("A confirmation email has been sent to email@example.com")
+        # Then I get redirect to the confirmation email send page with the email pre-filled
+        self.assertInUrl("/submitted/confirmation-email/send")
+        self.assertInBody("Send a confirmation email")
+        self.assertInBody("email@example.com")
 
-    def test_thank_you_page_confirmation_email_show_warning(self):
+    def test_confirm_email_yes(self):
         # Given I launch and complete the test_confirmation_email questionnaire
         self._launch_and_complete_questionnaire()
 
-        # When I enter a valid email and submit
+        # When I enter a valid email submit and answer yes on the confirm email page
         self.post({"email": "email@example.com"})
+        self.post({"confirm-email": "Yes, send the confirmation email"})
 
-        # Then I get a sign out warning
+        # Then I get confirmation that the email has been sent
         self.assertInUrl("confirmation-email/sent")
         self.assertInBody(
             'Make sure you <a href="/sign-out">leave this page</a> or close your browser if using a shared device'
         )
 
+    def test_confirm_email_confirmation_email_limit_reached(
+        self,
+    ):
+        # Given I launch and complete the test_confirmation_email questionnaire and reach the email confirmation limit
+        self._launch_and_complete_questionnaire()
+        self.post({"email": "email@example.com"})
+        self.last_url
+        self.post({"confirm-email": "Yes, send the confirmation email"})
+        self.get("/submitted/confirmation-email/send/")
+        self.post({"email": "email@example.com"})
+        confirm_email_url = self.last_url
+        self.post({"confirm-email": "Yes, send the confirmation email"})
+
+        # When I try to access the confirm email page
+        self.get(confirm_email_url)
+
+        # Then I get routed to the thank you page
+        self.assertInUrl("/submitted/thank-you/")
```

Comment:
This should be done for "/send" also. It could just be a subtest with the urls being the params. So when visiting `/send` or `/confirm`, they both should redirect to the thank you page.